Python/codegen.c

```diff
@@ -5254,14 +5277,33 @@ codegen_subscript(compiler *c, expr_ty e)
     return SUCCESS;
 }
 
-/* Returns the number of the values emitted,
- * thus are needed to build the slice, or -1 if there is an error. */
+/* Returns the number of the values emitted that are needed
+ * to build the slice. May be 0 if a constant slice was emitted.
+ * -1 if there is an error. */
 static int
 codegen_slice(compiler *c, expr_ty s)
 {
     int n = 2;
     assert(s->kind == Slice_kind);
 
+    if (is_constant_slice(s)) {
+        PyObject *start = NULL;
+        if (s->v.Slice.lower) {
+            start = s->v.Slice.lower->v.Constant.value;
+        }
+        PyObject *stop = NULL;
+        if (s->v.Slice.upper) {
+            stop = s->v.Slice.upper->v.Constant.value;
+        }
+        PyObject *step = NULL;
+        if (s->v.Slice.step) {
+            step = s->v.Slice.step->v.Constant.value;
+        }
+        PyObject *slice = PySlice_New(start, stop, step);
+        ADDOP_LOAD_CONST_NEW(c, LOC(s), slice);
+        return 0;
```

Comment:
Yeah, that makes sense.  What's weird about it is that the "net effect" of `codegen_slice` is different whether it's constant or not.

I factored out the part that just emits `start` and `stop`, since that's the part that's used in a few places, and then made `codegen_slice` always create a "ready to use" slice.  Hope that makes sense.